proxies/nodejs/handlers/location.ts

```diff
@@ -0,0 +1,140 @@
+import Koa from 'koa'
+import { DVCClient, DVCUser } from '@devcycle/nodejs-server-sdk'
+import { getEntityModelFromType, getEntityNameFromType } from '../entityTypes'
+
+//HTTP request comes in as string
+type LocationRequestBody = {
+  command: string;
+  params: string; //[{value: string | number | boolean} | {location: string} | {callbackUrl: string}] 
+  isAsync: boolean;
+};
+
+export const handleLocation = async (
+    ctx: Koa.ParameterizedContext,
+    data: {
+    clients: { [key: string]: DVCClient };
+    users: { [key: string]: DVCUser };
+    commandResults: { [key: string]: any };
+  }
+) => {
+    const body = ctx.request.body as LocationRequestBody
+    const entity = getEntityFromLocation(ctx.request.url, data)
+
+    if (entity === undefined) {
+        ctx.status = 400
+        ctx.body = {
+            errorCode: 400,
+            errorMessage: 'Invalid request: missing entity',
+        }
+    } else if (body.command === undefined) {
+        ctx.status = 400
+        ctx.body = {
+            errorCode: 400,
+            errorMessage: 'Invalid request: missing command',
+        }
+    } else {
+        try {
+            const command = body.command
+            const params: any | string | boolean | number = parseParams(
+                JSON.parse(body.params),
+                data
+            )
+
+            // TODO: handle async commands
+            // TODO: handle callback before invoking command
+            const resultData = await invokeCommand(
+                entity,
+                command,
+                params,
+                body.isAsync
+            )
+
+            const entityType = getEntityModelFromType(resultData.constructor.name)
+            const entityName = getEntityNameFromType(resultData.constructor.name)
+
+            const commandId = data.commandResults[entityName] !== undefined?
+                Object.keys(data.commandResults[entityName]).length :
+                0
+
+            if (data.commandResults[entityName] === undefined) {
+                data.commandResults[entityName] = {}
```

Comment:
we can probably just define the data object with the commands it can store so we don't have to do this

Reply:
that would only help if we are explicitly calling certain commands such as
 `user` and `variable` `onUpdate` `onInitialized`
 
 this model would allow for better future proofing?